src/coprocessor/dag/batch_executor/table_scan_executor.rs

```diff
@@ -310,482 +312,296 @@ mod tests {
     use crate::coprocessor::dag::expr::EvalConfig;
     use crate::coprocessor::util::convert_to_prefix_next;
     use crate::storage::{FixtureStore, Key};
-    use cop_datatype::FieldTypeAccessor;
-    use cop_datatype::FieldTypeTp;
+    use cop_datatype::{EvalType, FieldTypeAccessor, FieldTypeTp};
     use kvproto::coprocessor::KeyRange;
     use std::sync::Arc;
     use tipb::expression::FieldType;
     use tipb::schema::ColumnInfo;
 
-    #[test]
-    fn test_basic() {
-        // Table Schema: ID (INT, PK), Foo (INT), Bar (FLOAT, Default 4.5)
-        // Column id:    1,            2,         4
-
-        const TABLE_ID: i64 = 7;
+    /// Table Schema: ID (INT, PK), Foo (INT), Bar (FLOAT, Default 4.5)
+    /// Column id:    1,            2,         4
+    struct TableScanTestWrapper {
+        // ID(INT,PK), Foo(INT), Bar(Float,Default 4.5)
+        pub data: Vec<(i64, Option<i64>, Option<f64>)>,
+        pub table_id: i64,
+        pub columns_info: Vec<ColumnInfo>,
+        pub store: FixtureStore,
+    }
 
-        // [(row_id, columns)] where each column: (column id, datum)
-        let data = vec![
-            (
-                1,
-                vec![
-                    // A full row.
-                    (2, Datum::I64(10)),
-                    (4, Datum::F64(5.2)),
-                ],
-            ),
-            (
-                3,
-                vec![
-                    (4, Datum::Null),
-                    // Bar column is null, even if default value is provided the final result
-                    // should be null.
-                    (2, Datum::I64(-5)),
-                    // Orders should not matter.
-                ],
-            ),
-            (
-                4,
-                vec![
-                    (2, Datum::Null),
-                    // Bar column is missing, default value should be used.
-                ],
-            ),
-            (
-                5,
-                vec![
-                    // Foo column is missing, NULL should be used.
-                    (4, Datum::F64(0.1)),
-                ],
-            ),
-            (
-                6,
-                vec![
-                    // Empty row
-                ],
-            ),
-        ];
+    impl Default for TableScanTestWrapper {
```

Comment:
Maybe just `new()`? By using default, we usually expect that something can be different (and this provides the default case) but here there is no chance for it to be something different here.